pkg/sources/git/unit.go

```diff
@@ -24,7 +27,33 @@ type SourceUnit struct {
 
 // Implement sources.SourceUnit interface.
 func (u SourceUnit) SourceUnitID() string {
-	return u.ID
+	return fmt.Sprintf("%s:%s", u.Kind, u.ID)
+}
+
+// Provide a custom Display method.
+func (u SourceUnit) Display() string {
+	switch u.Kind {
+	case UnitRepo:
+		repo := u.ID
+		if parsedURL, err := url.Parse(u.ID); err == nil {
+			// scheme://host/owner/repo
+			repo = strings.TrimPrefix(parsedURL.Path, "/")
+		} else if _, path, found := strings.Cut(u.ID, ":"); found {
+			// git@host:owner/repo
+			// TODO: Is this possible? We should maybe canonicalize
+			// the URL before getting here.
+			repo = path
+		}
+		return strings.TrimSuffix(repo, ".git")
+	case UnitDir:
+		return filepath.Base(u.ID)
+	default:
+		return "mysterious git unit"
```

Comment:
This is theoretically possibly user-facing  - are you happy with the text?

Reply:
Yes, lol. Open to suggestions though :)